File: my-folder/0199-binary-tree-right-side-view/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    void f(TreeNode* root,vector<int> &v,int maxEle){
        if(root == NULL) return;
        
        if(v.size() <=maxEle){
            v.push_back(root->val);
            
           
        }
        
        f(root ->right,v,maxEle+1);
        f(root->left,v,maxEle+1);
      
   }
    vector<int> rightSideView(TreeNode* root) {
        vector<int> v;
        int maxEle = 0;
        f(root,v,maxEle);
        return v;
        
    }
};

File: my-folder/0199-binary-tree-right-side-view/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    void solve(TreeNode root,int level,List<Integer> ans){
        if(root == null)return;

        if(level == ans.size()){
            int sz = ans.size();
            ans.add(sz,root.val );
        }

        solve(root.right,level + 1,ans);
        solve(root.left,level + 1,ans);
    }

    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans = new ArrayList<Integer>();

        solve(root,0,ans);
        return ans;
    }
}
